modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/ObjectRelationship/ObjectRelationship.js

```diff
@@ -12,44 +12,130 @@
  * details.
  */
 
-import Autocomplete from 'commerce-frontend-js/components/autocomplete/Autocomplete';
+import ClayAutocomplete from '@clayui/autocomplete';
+import {useResource} from '@clayui/data-provider';
+import ClayDropDown from '@clayui/drop-down';
+import {useDebounce} from '@clayui/shared';
 import {FieldBase} from 'dynamic-data-mapping-form-field-type/FieldBase/ReactFieldBase.es';
+import {fetch} from 'frontend-js-web';
 import React from 'react';
 
-const ObjectRelationship = ({
+const NETWORK_STATUS_LOADING = 1;
+const NETWORK_STATUS_UNUSED = 4;
+
+const LoadingWithDebounce = ({loading, render}) => {
+	const debouncedLoadingChange = useDebounce(loading, 500);
+
+	if (loading || debouncedLoadingChange) {
+		return (
+			<ClayDropDown.Item className="disabled">
+				{Liferay.Language.get('loading')}
+			</ClayDropDown.Item>
+		);
+	}
+
+	return render;
+};
+
+export function ObjectRelationship({
 	apiURL,
-	initialLabel,
-	initialValue,
+	initialLabel = '',
+	initialValue = '',
 	inputName,
-	itemsKey,
-	itemsLabel,
+	labelKey = 'label',
 	name,
-	onChange,
 	readOnly,
 	required,
-	value,
+	valueKey = 'value',
 	...otherProps
-}) => (
-	<FieldBase
-		name={name}
-		readOnly={readOnly}
-		required={required}
-		{...otherProps}
-	>
-		<Autocomplete
-			apiUrl={apiURL}
-			initialLabel={initialLabel}
-			initialValue={initialValue}
-			inputName={inputName}
-			itemsKey={itemsKey}
-			itemsLabel={itemsLabel}
+}) {
+	const [selectedValue, setSelectedValue] = React.useState(initialValue);
```

Comment:
There is usually an idea that a Field's `value` state is bidirectional, it maintains an internal state and is also updated with some value received via `value` because this component is also used as a preview in the builder and is also used in Sidebar.

The Sidebar and the user view happens what is called evaluation, every time the user types a value in the field and calls `onChange` it makes a call to the backend to validate the properties of the field, there are other things that it validates too, it can happen the value will change and you will have to update with what came from the backend. But for this case, I think it doesn't exist.

Usually, the `useSyncValue` hook is used which will take care of this.